src/Processors/QueryPipeline.cpp

```diff
@@ -259,9 +264,50 @@ void QueryPipeline::addSimpleTransform(const ProcessorGetterWithStreamKind & get
     addSimpleTransformImpl(getter);
 }
 
+void QueryPipeline::setSinks(const ProcessorGetterWithStreamKind & getter)
```

Comment:
What is `sink`? Is it like `ResizingProcessor`, which resizes many to one?

Reply:
Actually, sink here is a processor with single input and no outputs.
This function create sink for each output port (via callback).
I will add comment.